Classes/Detector/OutOfOfficeDetector.php

```
@@ -146,20 +136,82 @@ protected function getBlockedPeriodType(DateTime $time, array $configuration): s
+    {
+        return 1 === preg_match('/^\d{4}-\d{2}-\d{2}$/', $pattern);
+    }
+
+    private function isMonthDayPattern(string $pattern): bool
+    {
+        return 1 === preg_match('/^\d{2}-\d{2}$/', $pattern);
+    }
+
+    private function isDateInRecurringRange(string $date, string $startPattern, string $endPattern): bool
+    {
+        [$year, $monthDay] = explode('-', $date, 2);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$year'. (undefined)

(UnusedLocalVariable)